### ISSUE TITLE
Client registration with arbitrary non-restricted / unknown scopes

### ISSUE THREAD

**Issue by vokac** [indigo-iam/iam#546]
Any **user** can register client with **arbitrary** non-restricted (unknown) **scope name** like `my_personal_scope`. I see several issues with this behavior
* difficult to catch misspelled scope names and pollute IAM database with scope names that doesn't make sense
* dangerous behavior in case we'll in future decide to use new restricted scopes and some users already registered these scope names

Our users are able to register client with e.g. `storage.write:/` scope (defined for SciTokens profile) and although this token have no privileges on the storage that comply with WLCG JWT profile it is at least quite confusing that such registration can be finished without filtering out unknown scopes.

Basically it is not clear to me why IAM allows unknown scope registration?

**Comment by vokac** [comment]
During WLCG BDT meeting it was mentioned new scopes can be useful for application developers, but still people generally agreed that we should not allow users to define arbitrary scope names and scopes defined by non-privileged IAM user should e.g. get a prefix, e.g. `test:my_personal_scope`

**Comment by giacomini** [comment]
The second bullet in the original post applies to https://github.com/indigo-iam/iam/pull/562, which adds a couple of scopes (`iam:admin:read` and `iam:admin:write`) meant to be restricted, hoping that a VO admin or user hasn't already added them. As a starting point to address the matter we would like to consider the `iam` prefix as a reserved namespace and later extend the approach to other situations. What do you think?

A side issue: at the moment we have chosen `:` as the separator character, but this is also the separator between the scope name and the parameter in case of parametric scopes (such as `storage.read:/some/path`). Is `.` a better alternative (such as in `wlcg.groups`)?

**Comment by vokac** [comment]
Reserved prefix `iam` prefix sounds reasonable and it doesn't collide with WLCG JWT or [AARC](https://docs.egi.eu/providers/check-in/sp/#claims). It seems to me that `.` is specific for WLCG/SCITOKEN, so I don't have strong opinion for `:` vs. `.` (it seems to me that `:` is used more often, e.g. [here](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes) and it the past this separator even had special meaning in [MitreId implementation](https://github.com/mitreid-connect/OpenID-Connect-Java-Spring-Server/issues/384))

**Comment by rmiccoli** [comment]
After reading these best practices about scopes (https://curity.io/resources/learn/scope-best-practices/), we agreed to use `iam:admin.read` and `iam:admin.write`

**Comment by federicaagostini** [comment]
PR https://github.com/indigo-iam/iam/pull/562
